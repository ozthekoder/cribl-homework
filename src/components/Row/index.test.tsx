import { render } from "@testing-library/react";
import { Row } from ".";
import "@testing-library/jest-dom";

describe("Row Component", () => {
  test("should render correctly with default styles", () => {
    const { container } = render(<Row>Test Content</Row>);
    expect(container.firstChild).toHaveStyle({
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-between",
      alignItems: "flex-start",
      height: "100%",
      width: "100%",
    });
  });

  test("should apply height correctly when passed as a prop", () => {
    const { container } = render(<Row height="50px">Test Content</Row>);
    expect(container.firstChild).toHaveStyle({
      height: "50px",
    });
  });

  test("should render children correctly", () => {
    const { getByText } = render(<Row>Child Element</Row>);
    expect(getByText("Child Element")).toBeInTheDocument();
  });

  test("should match snapshot", () => {
    const { container } = render(<Row height="75px">Snapshot Test</Row>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
