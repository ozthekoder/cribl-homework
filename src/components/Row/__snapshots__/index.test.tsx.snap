// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component should match snapshot 1`] = `
<div
  class="css-17w3u9e"
  height="75px"
>
  Snapshot Test
</div>
`;
